src/db_helpers.rs

```diff
@@ -44,24 +45,73 @@ impl TryFrom<&BlockView> for Block {
     }
 }
 
-pub(crate) async fn get_block_from_params(
-    pool: &sqlx::Pool<sqlx::Postgres>,
-    params: &types::query_params::BlockParams,
+pub(crate) fn timestamp_to_event_index(timestamp: u64) -> u128 {
+    timestamp as u128 * 100_000_000 * 100_000_000
+}
+
+pub(crate) fn event_index_to_timestamp(event_index: u128) -> u64 {
+    (event_index / (100_000_000 * 100_000_000)) as u64
+}
+
+pub(crate) async fn get_block_from_pagination(
+    pool_explorer: &sqlx::Pool<sqlx::Postgres>,
+    pagination: &types::query_params::Pagination,
 ) -> crate::Result<Block> {
-    if let Some(block_height) = params.block_height {
+    if let Some(event_index) = pagination.after_event_index {
+        checked_get_block(
+            pool_explorer,
+            &types::query_params::BlockParams {
+                block_timestamp_nanos: Some(event_index_to_timestamp(event_index).into()),
+                block_height: None,
+            },
+        )
+        .await
+    } else {
+        checked_get_block(
+            pool_explorer,
+            &types::query_params::BlockParams {
+                block_timestamp_nanos: None,
+                block_height: None,
+            },
+        )
+        .await
+    }
```

Comment:
```suggestion
        checked_get_block(
            pool_explorer,
            &types::query_params::BlockParams {
                block_timestamp_nanos: pagination
                    .after_event_index
                    .map(|event_index| event_index_to_timestamp(event_index).into()),
                block_height: None,
            },
        )
        .await
```
The only difference between these blocks is what you pass to `block_timestamp_nanos`, so we can just inline the conditional to make it more clear. This also allows us to `map` the `Option` :).

What do you think?

Reply:
Also why is `block_height` always `None`? Is this some capability that will be added later?

Reply:
I agree, your version is better!
`block_height` can be `Some` when the user provides it